tests/smoke-tests/src/test/java/org/apache/activemq/artemis/tests/smoke/jms/ManifestTest.java

```diff
@@ -14,53 +14,58 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.activemq.artemis.tests.unit.jms.misc;
+package org.apache.activemq.artemis.tests.smoke.jms;
 
 import javax.jms.ConnectionMetaData;
 import java.io.File;
-import java.util.Properties;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import java.util.jar.Attributes;
 import java.util.jar.JarFile;
 import java.util.jar.Manifest;
 
 import org.apache.activemq.artemis.core.server.ActiveMQServer;
 import org.apache.activemq.artemis.core.server.ActiveMQServers;
+import org.apache.activemq.artemis.core.version.Version;
 import org.apache.activemq.artemis.jms.client.ActiveMQConnectionMetaData;
-import org.apache.activemq.artemis.tests.util.ActiveMQTestBase;
+import org.apache.activemq.artemis.tests.smoke.common.SmokeTestBase;
 import org.jboss.logging.Logger;
 import org.junit.Assert;
 import org.junit.Test;
 
-public class ManifestTest extends ActiveMQTestBase {
+public class ManifestTest extends SmokeTestBase {
 
    private static final Logger log = Logger.getLogger(ManifestTest.class);
-
-
+   private static List<String> jarFiles = new ArrayList<>(Arrays.asList(
+           "artemis-jms-client-", "artemis-jms-server-"));
 
    @Test
    public void testManifestEntries() throws Exception {
-      Properties props = System.getProperties();
-      String userDir = props.getProperty("build.lib");
-
-      log.trace("userDir is " + userDir);
-
-      // The jar must be there
-      File file = new File("build/jars", "activemq-core.jar");
-      Assert.assertTrue(file.exists());
 
-      // Open the jar and load MANIFEST.MF
-      JarFile jar = new JarFile(file);
-      Manifest manifest = jar.getManifest();
 
       ActiveMQServer server = ActiveMQServers.newActiveMQServer(createBasicConfig());
-
-      ConnectionMetaData meta = new ActiveMQConnectionMetaData(server.getVersion());
-
-      // Compare the value from ConnectionMetaData and MANIFEST.MF
-      Attributes attrs = manifest.getMainAttributes();
-
-      Assert.assertEquals(meta.getProviderVersion(), attrs.getValue("ActiveMQ-Version"));
+      Version serverVersion = server.getVersion();
+      String serverFullVersion = serverVersion.getFullVersion();
+      ConnectionMetaData meta = new ActiveMQConnectionMetaData(serverVersion);
+
+      String distributionDir = basedir + "/../../artemis-distribution/target/apache-artemis-" +
+         serverFullVersion + "-bin/apache-artemis-" + serverFullVersion;
```

Comment:
@gemmellr I made few changes as you suggested. Moved it from integration to smoke. No there is no need for the build.lib paramerter but I didn't find a way to get the artemis-distribution path. If I use the artemis-maven-plugin to generate a server the generated server folder has a profile file that points to the distrubution directory where I can look for the jars inside the lib directory. But I cannot find a way to retrieve this directory in the code. So, for now as you can see it is partially hardcoded. Any suggestions to void this hardcoded part?

Reply:
Not offhand no, I actually thought the smoke tests bits already had that value, but it seems like its actually only artemis-maven-plugin that does (perhaps from [1]) which in turn puts it in the generated profile file as you noted, with the smoke tests only knowing the path to the configs. So adding a 'distribution lib' style property as you had before may indeed be the simplest for now after all unfortunately.

[1] https://github.com/apache/activemq-artemis/blob/2.20.0/artemis-maven-plugin/src/main/java/org/apache/activemq/artemis/maven/ArtemisCreatePlugin.java#L66

Reply:
Ok, so I added the property in the smoke test pom. So, @brusdev @gemmellr let me know if any other change will be required.